cpe_workspaceone/resources/cpe_workspaceone.rb

```diff
@@ -43,22 +50,53 @@ def uninstall?
 
   def enforce_mdm_profiles
     return unless node['cpe_workspaceone']['mdm_profiles']['enforce']
+
     macos_enforce_mdm_profiles if node.macos?
   end
 
+  def set_cli_config(flag, val)
+    default = WS1_DEFAULT_PREFS[flag]
+    val = val || default
+    cmd = node.hubcli_execute("config --set #{flag} #{val}")
+    unless cmd.exitstatus.zero?
+      if !cmd.stderr.include?('Error: Invalid value for option') || val == default
+        cmd.error!
+      end
+      Chef::Log.warn("cpe_workspaceone - #{cmd.stderr.strip} (#{val}) - setting default")
+      set_cli_config(flag, default)
+    end
+  end
+
+  def manage_cli_config
+    unless node.ws1_hubcli_exists
+      Chef::Log.warn('cpe_workspaceone - hubcli path does not exist, cannot enforce MDM profiles!')
+      return
+    end
+
+    prefs = node['cpe_workspaceone']['cli_prefs'].reject { |_, v| v.nil? }
+    prefs.each do |flag, val|
+      unless WS1_DEFAULT_PREFS.keys.include?(flag)
+        Chef::Log.warn("cpe_workspaceone - refusing to manage unknown cli preference '#{flag}'")
+        next
+      end
+
+      set_cli_config(flag, val)
+    end
+  end
+
   def macos_enforce_mdm_profiles
     # Bail if path doesn't exist
     unless node.ws1_hubcli_exists
       Chef::Log.warn('cpe_workspaceone - hubcli path does not exist, cannot enforce MDM profiles!')
       return
     end
 
+    forcelist = node['cpe_workspaceone']['mdm_profiles']['profiles']['force'] || []
```

Comment:
There is only a single `forcelist` but it is referenced in both use and machine level profiles. This logic should be split out to reduce the calls to WS1 API/hubcli.

Reply:
Split and documented split off forcelist into `user_forced` and `device_forced`. It is not very clean, open to other ideas on how to structure this part of the attributes API, or perhaps throw it out.

Reply:
This is fine for now. I have some ideas on how to make this type of functionality better, but not necessarily how to make it cleaner. I think it is what it is for now.